modules/WebPublishPlugin/src/main/java/net/clementlevallois/web/publish/plugin/controller/JPanelWebExport.java

```diff
@@ -0,0 +1,612 @@
+
+    public JPanelWebExport() {
+        initComponents();
+        Preferences preferences = NbPreferences.forModule(this.getClass());
+        accessToken = preferences.get(ACCESS_TOKEN_KEY_IN_USER_PREFS, "");
+        if (accessToken == null || accessToken.isBlank()) {
+            jLabelAlreadyLoggedIn.setVisible(false);
+        } else {
+            jLabelAlreadyLoggedIn.setVisible(true);
+        }
+        jTextFieldGithubErrorMsg.setBackground(Color.WHITE);
+        jTextFieldGithubErrorMsg.setText(bundle.getString("general.message.errors_appear.here"));
+        jTextFieldGithubErrorMsg.setCaretPosition(0);
+        jTextFieldUserCode.setForeground(Color.RED);
+        jTextAreaUrls.setText("");
+    }
+
+    SwingWorker pollWorker = new SwingWorker<Void, Integer>() {
+
+        @Override
+        protected Void doInBackground() throws Exception {
+            JsonObject jsonObject = new JsonObject();
+            try {
+                HttpClient client = HttpClient.newHttpClient();
+                String url = "https://github.com/login/oauth/access_token";
+
+                String inputParams = "client_id="
+                        + GEPHI_APP_CLIENT_ID
+                        + "&"
+                        + "device_code="
+                        + deviceCode
+                        + "&"
+                        + "grant_type=urn:ietf:params:oauth:grant-type:device_code";
+                HttpRequest request = HttpRequest.newBuilder()
+                        .uri(URI.create(url))
+                        .header("accept", "application/json")
+                        .POST(HttpRequest.BodyPublishers.ofString(inputParams))
+                        .build();
+
+                boolean success = false;
+                long startTime = System.currentTimeMillis();
+                long maxDuration = 900_000;
+                float currDuration = 0;
+                int loops = 0;
+                while (!success && currDuration < maxDuration) {
+                    HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+                    JsonElement responseAsJsonElement = JsonParser.parseString(response.body());
+                    jsonObject = responseAsJsonElement.getAsJsonObject();
+                    if (jsonObject.has("access_token")) {
+                        break;
+                    }
+                    currDuration = (float) (System.currentTimeMillis() - startTime) / (float) 1000;
+                    // the min duration recommended by Github between two polls is 5 seconds
+                    // -> https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps#device-flow
+                    Thread.sleep(5200);
+                    publish(++loops);
+                }
+            } catch (IOException | InterruptedException ex) {
+                Exceptions.printStackTrace(ex);
+                jTextFieldGithubErrorMsg.setText(ex.getMessage());
```

Comment:
Within the `SwingWorker` class you're running on a separate thread than the EDT (the swing thread). All the Swing code should normally be sent back to the EDT, usually with `SwingUtilities.invokeLater()`. It will work but not fully correct.